src/Controller/Api/ApiController.php

```diff
@@ -2,107 +2,81 @@
 
 namespace App\Controller\Api;
 
-use Symfony\Component\HttpFoundation\{Request, Response};
-use Symfony\Component\HttpKernel\Exception\{NotFoundHttpException, UnsupportedMediaTypeHttpException};
-use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\HttpFoundation\{JsonResponse, Request};
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Routing\{Annotation\Route, Generator\UrlGeneratorInterface};
+use Symfony\Contracts\Translation\TranslatorInterface;
 
 class ApiController
 {
     /**
-     * Returns information about client and writes it to log file.
+     * Returns all resources of application.
      *
-     * @Route("/api/my-info", methods="GET", name="client_info")
+     * @Route("/api", methods="GET", name="list_of_resources")
      *
-     * @param Request $request
+     * @param UrlGeneratorInterface $urlGenerator
      *
-     * @return Response
+     * @return JsonResponse
      */
-    public function actionClientInfo(Request $request): Response
-    {
-        if (!$request->headers->get('Content-type: application/json')) {
-            throw new UnsupportedMediaTypeHttpException();
-        }
-
-        $clientInfo = [
-            'ip' => $request->getClientIps()[0],
-            'language' => $request->getLanguages(),
-            'browser' => ($request->headers->get('User-Agent'))
-        ];
-        $responseData = json_encode($clientInfo);
-
-        file_put_contents('logs.txt', $responseData . PHP_EOL, FILE_APPEND | LOCK_EX);
+    public function actionIndex(
+        UrlGeneratorInterface $urlGenerator,
+    ): JsonResponse {
+        $data = $this->getDataFromFile();
 
-        return new Response(
-            $responseData,
-            Response::HTTP_OK,
-            [
-                'Content-type' => 'application/json'
-            ]
-        );
+        return new JsonResponse($this->generateUrls($data, $urlGenerator));
     }
 
     /**
-     * Returns all resources of application.
+     * Returns information about client and writes it to log file.
      *
-     * @Route("/api", methods="GET", name="list_of_resources")
+     * @Route("/api/my-info", methods="GET", name="client_info")
      *
      * @param Request $request
      *
-     * @return Response
+     * @return JsonResponse
      */
-    public function actionIndex(Request $request): Response
+    public function actionClientInfo(Request $request): JsonResponse
     {
-        if (!$request->headers->get('Content-type: application/json')) {
-            throw new UnsupportedMediaTypeHttpException();
-        }
+        $clientInfo = [
+            'ip' => $request->getClientIps()[0],
+            'language' => $request->getPreferredLanguage(),
+            'browser' => $request->headers->get('User-Agent')
+        ];
+        $encoded = json_encode($clientInfo);
```

Comment:
there is no need to encode an array, if it's rendered with the JsonResponse

Reply:
It's just for logging purposes which are at 47